CONTRIBUTING.md

```diff
@@ -0,0 +1,291 @@
+  able to work a lot faster as you build experience.
+
+### Working on an issue
+
+To work on an issue, claim it by adding a comment with `@zulipbot claim` to
+the issue thread. [Zulipbot](https://github.com/zulip/zulipbot) is a GitHub
+workflow bot; it will assign you to the issue and label the issue as "in
+progress". Some additional notes:
+
+* You're encouraged to ask questions on how to best implement or debug your
+  changes -- the Zulip maintainers are excited to answer questions to help
+  you stay unblocked and working efficiently. You can ask questions on
+  chat.zulip.org, or on the GitHub issue or pull request.
+* We encourage early pull requests for work in progress. Prefix the title of
+  work in progress pull requests with `[WIP]`, and remove the prefix when
+  you think it might be mergeable and want it to be reviewed.
+* After updating a PR, add a comment to the GitHub thread mentioning that it
+  is ready for another review. GitHub only notifies maintainers of the
+  changes when you post a comment, so if you don't, your PR will likely be
+  neglected by accident!
+
+### And beyond
+
+A great place to look for a second issue is to look for issues with the same
+`area:` label as the last issue you resolved. You'll be able to reuse the
+work you did learning how that part of the codebase works. Also, the path to
+becoming a core developer often involves taking ownership of one of these area
+labels.
+
+## What makes a great Zulip contributor?
+
+Zulip runs a lot of internship programs, so we have a lot of experience with
+new contributors. In our experience, these are the best predictors of success:
+
+* Posting good questions. This generally means explaining your current
+  understanding, saying what you've done or tried so far, and including
+  tracebacks or other error messages if appropriate.
+* Learning and practicing
+  [Git commit discipline](http://zulip.readthedocs.io/en/latest/version-control.html#commit-discipline).
+* Submitting carefully tested code. This generally means checking your work
+  through a combination of automated tests and manually clicking around the
+  UI trying to find bugs in your work. See
+  [things to look for](http://zulip.readthedocs.io/en/latest/code-reviewing.html#things-to-look-for)
+  for additional ideas.
+* Posting
+  [screenshots or GIFs](http://zulip.readthedocs.io/en/latest/screenshot-and-gif-software.html)
+  for frontend changes.
+* Being responsive to feedback on pull requests. This means incorporating or
+  responding to all suggested changes, and leaving a note if you won't be
+  able to address things within a few days.
+* Being helpful and friendly on chat.zulip.org.
+
+These are also the main criteria we use to select interns for all of our
+internship programs.
+
+## Reporting issues
+
+If you find an easily reproducible bug and/or are experienced in reporting
+bugs, feel free to just open an issue on the relevant project on GitHub.
```

Comment:
We have a ReadTheDocs article on reporting bugs; we should link to it.  `docs/bug-reports.md`.  It's not the best; the good content in there is the links.